indicators/views/bulk_indicator_import_views.py

```diff
@@ -0,0 +1,425 @@
+                        indicator_data['sector'] = -1
+                else:
+                    indicator_data[column['field_name']] = cell_value
+            indicator_data['level'] = current_level.id
+
+            new_indicators_data.append(indicator_data)
+
+        if len(new_indicators_data) == 0:
+            return JsonResponse({'error_code': ERROR_NO_NEW_INDICATORS}, status=400)
+
+        if len(workbook_errors) > 0:
+            return JsonResponse({'errors': workbook_errors}, status=400)
+
+        deserialized_indicators = BulkImportIndicatorSerializer(data=new_indicators_data, many=True)
+        deserialized_indicators.is_valid()
+        error_count = sum([len(error_dict) for error_dict in deserialized_indicators.errors])
+
+        if error_count > 0:
+            try:
+                old_file_obj = BulkIndicatorImportFile.objects.get(
+                    user=request.user.tola_user,
+                    program=program,
+                    file_type=BulkIndicatorImportFile.INDICATOR_DATA_TYPE)
+                os.remove(old_file_obj.file_path)
+                old_file_obj.delete()
+            except BulkIndicatorImportFile.DoesNotExist:
+                pass
+
+            file_name = f'{datetime.strftime(datetime.now(), "%Y%m%d-%H%M%S")}-{request.user.id}-{program.pk}.xlsx'
+            file_obj = BulkIndicatorImportFile.objects.create(
+                user=request.user.tola_user,
+                program=program,
+                file_name=file_name,
+                file_type=BulkIndicatorImportFile.INDICATOR_TEMPLATE_TYPE)
+            with open(file_obj.file_path, 'w') as fh:
+                wb.save(file_obj.file_path)
+            return JsonResponse({
+                'invalid': error_count,
+                'valid': len(deserialized_indicators.errors) - error_count,
+                'error_code': ERROR_MALFORMED_INDICATOR
+            }, status=400)
+        else:
+            try:
+                old_file_obj = BulkIndicatorImportFile.objects.get(
+                    user=request.user.tola_user,
+                    program=program,
+                    file_type=BulkIndicatorImportFile.INDICATOR_DATA_TYPE)
+                os.remove(old_file_obj.file_path)
+                old_file_obj.delete()
+            except BulkIndicatorImportFile.DoesNotExist:
+                pass
+
+            file_name = f'{datetime.strftime(datetime.now(), "%Y%m%d-%H%M%S")}-{request.user.id}-{program.pk}.json'
+            file_obj = BulkIndicatorImportFile.objects.create(
+                user=request.user.tola_user,
+                program=program,
+                file_name=file_name,
+                file_type=BulkIndicatorImportFile.INDICATOR_DATA_TYPE)
+            with open(file_obj.file_path, 'w') as fh:
+                fh.write(json.dumps(new_indicators_data))
+            return JsonResponse({'message': 'success', 'valid': len(new_indicators_data), 'invalid': 0}, status=200)
```

Comment:
Making sure I understand this correctly, we are keeping at most 1 redline feedback file per user id per program and when the request comes in for the redline feedback file, you are using the user id and program to get it? So we don't need to send/receive a user id to/from the front end.

Reply:
Correct about the just keeping one file per user per program.  The user id is inherently part of the request.  It is extracted by, I believe, the Django session middleware and provided as `request.user` to the Django view.  So we don't have to include it explicitly in the request (and indeed we wouldn't take a user id submitted in that way at face value since anyone could formulate a request with a someone else's user id).